hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -337,6 +346,13 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
         // So that this key can't be committed again.
         omMetadataManager.getOpenKeyTable(getBucketLayout()).addCacheEntry(
             dbOpenKey, trxnLogIndex);
+
+        // Prevent hsync metadata from getting committed to the final key
+        omKeyInfo.getMetadata().remove(OzoneConsts.HSYNC_CLIENT_ID);
+      } else if (updateOpenKeyTable) {
+        // Both isHSync and updateOpenKeyTable are true, update OpenKeyTable
+        omMetadataManager.getOpenKeyTable(getBucketLayout()).addCacheEntry(
+            dbOpenKey, omKeyInfo, trxnLogIndex);
```

Comment:
You cannot use the same omKeyInfo for keyTable and openKeyTable. There is one case, that a file's all blocks are per-allocated. Then the openKeyInfo will have all these blocks info, while the keyInfo only has the hsynced block info.

Reply:
Good catch. Let me add the hsync metadata tag solely without touching any other properties then.

Reply:
done in 7bdb08312a11ca9c37114eee0840be9a96460a01